scalyr_agent/scalyr_monitor.py

```diff
@@ -740,28 +766,56 @@ def get(self, field, required_field=False, max_value=None, min_value=None,
             present.
         @param convert_to: If not None, then will convert the value for the field to the specified type. Only int,
             bool, float, long, str, and unicode are supported. If the type conversion cannot be done, a
-            BadMonitorConfiguration exception is raised. The only true conversions allowed are those from str, unicode
-            value to other types such as int, bool, long, float. Trivial conversions are allowed from int, long to
+            BadMonitorConfiguration exception is raised. The only conversions allowed are those mapped out in
+            ALLOWED_CONVERSIONS. Trivial conversions are allowed from int, long to
             float, but not the other way around. Additionally, any primitive type can be converted to str, unicode.
+            str, unicode can be converted to complex types such as ArrayOfStrings, JsonArray, JsonObject as long as
+            they can be correctly parsed.
         @param default: The value to return if the field is not present in the configuration. This is ignored if
             'required_field' is True.
         @param max_value: If not None, the maximum allowed value for field. Raises a BadMonitorConfiguration if the
             value is greater.
         @param min_value: If not None, the minimum allowed value for field. Raises a BadMonitorConfiguration if the
             value is less than.
+        @param report_conflicting_environment_value: If True, disallows overriding via environment variable.
 
         @return: The value
         @raise BadMonitorConfiguration: If any of the conversion or required rules are violated.
         """
-        if required_field and field not in self.__map:
-            raise BadMonitorConfiguration('Missing required field "%s"' % field, field)
-        result = self.__map.get(field, default)
+        result = self.__map.get(field)
+        if result is not None and convert_to is not None and type(result) != convert_to:
+            result = self.__perform_conversion('config param', field, result, convert_to)
+
+        # Param not found in config file, so check environment
+        if result is None or report_conflicting_environment_value:
+            if hasattr(self.__monitor_class, 'ENVIRONMENT_AWARE_CONFIG_PARAMS'):
+                env_aware_param_map = self.__monitor_class.ENVIRONMENT_AWARE_CONFIG_PARAMS
+                envar_name, envar_strval, envar_val = get_config_from_env(
+                    field, env_aware_param_map.get(field), convert_to=convert_to
+                )
+                if result is None:
+                    result = envar_val
+                elif report_conflicting_environment_value and envar_val:
+                    # Both config param and environment variables defined.  Warn if conflict.
+                    if result != envar_val and log:
+                        log.warn(
```

Comment:
Same here about the using the `limit_once_per_x_secs` feature to limit to once per 5 minutes.